File: Contribute/Plugin Samples/Ruby/WavyStylePlugin/Form1.rb
require "mscorlib"
require "System, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "System.Collections.Generic, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "System.ComponentModel, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "System.Data, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "System.Drawing, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "System.Linq, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "System.Text, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "System.Windows.Forms, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "SS.Ynote.Engine.Framework.Plugins.Interface, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "System.Xml.Linq, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "FastColoredTextBoxNS, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "SS.Ynote.Classic, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
#========================================
#
#Wavy Style can be use in error checking
#you can make a plugin for error checking for any language
#
#This is just a sample
#
#========================================
module WavyStylePlugin
    # <summary>
    # Simple Spell Checker
    # </summary>
    class Form1 < WeifenLuo::WinFormsUI::Docking::DockContent, IFormPlugin
        def initialize()
            @RedWavy = WavyLineStyle.new(100, Color.Red)
            @configuration = XElement.new("WavyStylePluginConfig")
            self.InitializeComponent()
            self.Hide()
            self.@Visible = false
            Main.ActiveEditor.codebox.TextChangedDelayed { |sender, e| codebox_TextChangedDelayed(sender, e) }
            MessageBox.Show("Spell Checker Initialized!")
        end

        def Title
            return "WavyStylePlugin"
        end

        def Description
            return "Wavy Style Sample Plugin"
        end

        def Group
            return "Plugins"
        end

        def SubGroup
            return "Samples"
        end

        def Configuration
            return @configuration
        end

        def Configuration=(value)
            @configuration = value
        end

        def Icon
            #get { return "C:\\Icons\\youricon.ico"; }
            return System::String.Empty
        end

        def Content
            return self
        end

        def ShowAs
            return self.ShowAs.Normal
        end

        def codebox_TextChangedDelayed(sender, e)
            SpellChecker = SpellChecker.new(Main.ActiveFastColoredTextBox, @"Dictionary.dic")
            SpellChecker.SpellCheck(sender)
        end
    end
end

File: Contribute/Plugin Samples/IronPython/WavyStylePlugin/Form1.py
from System import *
from System.Collections.Generic import *
from System.ComponentModel import *
from System.Data import *
from System.Drawing import *
from System.Linq import *
from System.Text import *
from System.Windows.Forms import *
from SS.Ynote.Engine.Framework.Plugins.Interface import *
from System.Xml.Linq import *
from FastColoredTextBoxNS import *
from SS.Ynote.Classic import *
#========================================
#
#Wavy Style can be use in error checking
#you can make a plugin for error checking for any language
#
#This is just a sample
#
#========================================
class Form1(WeifenLuo.WinFormsUI.Docking.DockContent, IFormPlugin):
    """ <summary>
     Simple Spell Checker
     </summary>
    """
    def __init__(self):
        self._RedWavy = WavyLineStyle(100, Color.Red)
        self._configuration = XElement("WavyStylePluginConfig")
        self.InitializeComponent()
        self.Hide()
        self._Visible = False
        Main.ActiveEditor.codebox.TextChangedDelayed += self.codebox_TextChangedDelayed
        MessageBox.Show("Spell Checker Initialized!")

    def get_Title(self):
        return "WavyStylePlugin"

    Title = property(fget=get_Title)

    def get_Description(self):
        return "Wavy Style Sample Plugin"

    Description = property(fget=get_Description)

    def get_Group(self):
        return "Plugins"

    Group = property(fget=get_Group)

    def get_SubGroup(self):
        return "Samples"

    SubGroup = property(fget=get_SubGroup)

    def get_Configuration(self):
        return self._configuration

    def set_Configuration(self, value):
        self._configuration = value

    Configuration = property(fget=get_Configuration, fset=set_Configuration)

    def get_Icon(self):
        #get { return "C:\\Icons\\youricon.ico"; }
        return str.Empty

    Icon = property(fget=get_Icon)

    def get_Content(self):
        return self

    Content = property(fget=get_Content)

    def get_ShowAs(self):
        return self.ShowAs.Normal

    ShowAs = property(fget=get_ShowAs)

    def codebox_TextChangedDelayed(self, sender, e):
        SpellChecker = SpellChecker(Main.ActiveEditor.codebox, @"Dictionary.dic")
        SpellChecker.SpellCheck(sender)